File: common/src/main/java/dev/niostone/interstellar/config/StarSystemsConfig.java
package dev.niostone.interstellar.config;

import me.shedaniel.autoconfig.ConfigData;
import me.shedaniel.autoconfig.annotation.Config;
import me.shedaniel.autoconfig.annotation.ConfigEntry;

@Config(name = "systems")
public class StarSystemsConfig implements ConfigData {
    @ConfigEntry.Category("sol")
    @ConfigEntry.Gui.TransitiveObject
    public StarSystemConfig sol;

    @ConfigEntry.Category("keplerBelt")
    @ConfigEntry.Gui.TransitiveObject
    public StarSystemConfig keplerBelt;

    @ConfigEntry.Category("proximaCentauri")
    @ConfigEntry.Gui.TransitiveObject
    public StarSystemConfig proximaCentauri;

    @ConfigEntry.Category("tauCeti")
    @ConfigEntry.Gui.TransitiveObject
    public StarSystemConfig tauCeti;

    @ConfigEntry.Category("gliese892")
    @ConfigEntry.Gui.TransitiveObject
    public StarSystemConfig gliese892;
}

File: common/src/main/kotlin/dev/niostone/interstellar/impl/config/StarSystemsConfig.kt
package dev.niostone.interstellar.impl.config

import dev.niostone.interstellar.impl.config.defaults.DefaultStarSystemConfigs
import me.shedaniel.autoconfig.ConfigData
import me.shedaniel.autoconfig.annotation.Config
import me.shedaniel.autoconfig.annotation.ConfigEntry

@Config(name = "systems")
class StarSystemsConfig: ConfigData {
    @ConfigEntry.Category("sol")
    @ConfigEntry.Gui.CollapsibleObject
    val sol = DefaultStarSystemConfigs.SOL

//    @ConfigEntry.Category("keplerBelt")
//    @ConfigEntry.Gui.CollapsibleObject
//    val keplerBelt
//
//    @ConfigEntry.Category("proximaCentauri")
//    @ConfigEntry.Gui.CollapsibleObject
//    val proximaCentauri
//
//    @ConfigEntry.Category("tauCeti")
//    @ConfigEntry.Gui.CollapsibleObject
//    val tauCeti
//
//    @ConfigEntry.Category("gliese892")
//    @ConfigEntry.Gui.CollapsibleObject
//    val gliese892
}
